NeoML/src/Dnn/ReferenceDnnFactory.cpp

```diff
@@ -118,18 +121,32 @@ void CReferenceDnnFactory::serialize( CArchive& archive, bool optimizeDnn )
 
 	NeoAssert( archive.IsLoading() );
 	Origin->Dnn.Serialize( archive );
-	archive.Close();
 
 	if( optimizeDnn == true ) {
+		NeoAssert( !Origin->Dnn.IsReferenceDnn() );
```

Comment:
ReferenceDnn cannot be serialized, can it? How can this condition fail? Maybe then we should check it always?

Reply:
Yes, ReferenceDnn cannot be serialized
This assert is here for ensure this method's trick still work correctly

Reply:
_Origin_ can become reference only if something wrong happened in _Serialize()_, and it is definitely some error. What I don't like here is that you are asserting it if _optimizeDnn_ and ignore otherwise. 
BTW, redundant `== true` here too.

Reply:
ok, fixed